app/api/csv/csvExporter.ts

```diff
@@ -0,0 +1,217 @@
+import { Writable } from 'stream';
+import { EventEmitter } from 'events';
+import * as csv from '@fast-csv/format';
+import templates from 'api/templates';
+import { PropertySchema } from 'shared/types/commonTypes';
+import { TemplateSchema } from 'shared/types/templateType';
+import translate, { getLocaleTranslation, getContext } from 'shared/translate';
+import translations from 'api/i18n/translations';
+import formatters from './typeFormatters';
+
+export type SearchResults = {
+  rows: any[];
+  totalRows: number;
+  aggregations: {
+    all: {
+      _types: {
+        buckets: {
+          key: string;
+          // eslint-disable-next-line camelcase
+          doc_count: number;
+          // eslint-disable-next-line camelcase
+          filtered: { doc_count: number };
+        }[];
+      };
+    };
+  };
+};
+
+export type ExporterOptions = {
+  dateFormat: string;
+  language: string;
+};
+
+export type ExportHeader = {
+  label: string;
+  name: string;
+  common: boolean;
+};
+
+export type TemplatesCache = {
+  [id: string]: TemplateSchema;
+};
+
+export const getTypes = (searchResults: SearchResults, typesWhitelist: string[] = []) =>
+  typesWhitelist.length > 0
```

Comment:
Unnecessary compare operator here: `typesWhitelis.length` is enough, as it will be true if > 0 and false if not.